src/dotnet/commands/dotnet-list/dotnet-list-tool/ListToolCommand.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Cli.CommandLine;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Configurer;
+using Microsoft.DotNet.ToolPackage;
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.DotNet.Tools.List.Tool
+{
+    internal class ListToolCommand : CommandBase
+    {
+        private const string CommandDelimiter = ", ";
+        private readonly AppliedOption _options;
+        private readonly IToolPackageStore _toolPackageStore;
+        private readonly IReporter _reporter;
+        private readonly IReporter _errorReporter;
+
+        public ListToolCommand(
+            AppliedOption options,
+            ParseResult result,
+            IToolPackageStore toolPackageStore = null,
+            IReporter reporter = null)
+            : base(result)
+        {
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+            _toolPackageStore = toolPackageStore ?? new ToolPackageStore(
+                new DirectoryPath(new CliFolderPathCalculator().ToolsPackagePath));
+            _reporter = reporter ?? Reporter.Output;
+            _errorReporter = reporter ?? Reporter.Error;
+        }
+
+        public override int Execute()
+        {
+            if (!_options.ValueOrDefault<bool>("global"))
+            {
+                throw new GracefulException(LocalizableStrings.ListToolCommandOnlySupportsGlobal);
+            }
+
+            var table = new PrintableTable<IToolPackage>();
+
+            table.AddColumn(
```

Comment:
There is no test coverage from 47 - 59. And it is pretty complex. I feel given the current code structure, it might be hard to test. You may need to extract a class out. All these logic are just given a `IToolPackage` and see the table out as a string. Maybe you can stub the result of `_toolPackageStore.GetInstalledPackages()` to be the input of you want

Reply:
There's no test coverage at all for the list tool comand yet (see WIP note on PR).